### PR TITLE
Document (active_)?face_iterator typedefs.

### PR DESCRIPTION
**Pull request by bangerth**
Part of #316.

### CODE REVIEW

File: include/deal.II/dofs/dof_handler.h [hunk 1]
```diff
@@ -256,7 +256,35 @@ class DoFHandler  :  public Subscriptor
...
+   * The concept of iterators is discussed at length in the
+   * @ref Iterators "iterators documentation module".
+   *
+   * While the actual data type of the typedef is hidden behind a few layers
+   * of (unfortunately necessary) indirections, it is in essence
+   * TriaIterator<DoFAccessor>. The
+   * TriaIterator class works like a pointer to objects that when
+   * you dereference it yields an object of type DoFAccessor. DoFAccessor,
+   * in turn, is a class that can be used to query DoF indices on faces,
+   * but it is also derived from TriaAccessor and consequently be used
```
Comment (davydden) [inline include/deal.II/dofs/dof_handler.h]: and consequently **can** be used

File: include/deal.II/dofs/dof_handler.h [hunk 2]
```diff
@@ -256,7 +256,35 @@ class DoFHandler  :  public Subscriptor
...
+   * to query geometric properties such as their vertices, their area, etc.
```
Comment (davydden) [inline include/deal.II/dofs/dof_handler.h]: maybe `..such as vertices of faces, their area...` as readers probably already forgotten what `their` is.

File: include/deal.II/hp/dof_handler.h [hunk 3]
```diff
@@ -237,8 +237,35 @@ namespace hp
     typedef TriaIterator < dealii::DoFCellAccessor < DoFHandler < dim, spacedim >, false > >        cell_iterator;
 #endif
 
-
+    /**
+     * A typedef that is used to identify iterators that point to faces.
```
Comment (davydden) [inline include/deal.II/hp/dof_handler.h]: maybe use `@copydoc` instead of copy-pastng?

**Comment by bangerth** [comment]
I fixed the grammar issues and used `@copydoc` where appropriate -- good idea!

**Pull request merged**